index.rst

```diff
@@ -615,6 +615,24 @@ data which is updated is small enough and will not require the scalable
 architecture – we plan to handle all Level 1 data set with out-of-the
 box MySQL as described in :ref:`alert-production`.
 
+.. _async-queries:
+
+Long-running queries
+~~~~~~~~~~~~~~~~~~~~
+
+Many of the typical user queries may need significant time to complete,
+at the scale of hours. To avoid re-submission of those long-running queries
+in case of various failures (networking or hardware issues) the system will
+support asynchronous query execution mode. In this mode users will submit
+queries using special options or syntax and the system will dispatch a
+query and immediately return to user some identifier of the submitted query
+without blocking user session. This query identifier will be used by user
+to retirieve query processing status, query result after query completes,
+or a partial query result while query is still executing.
```

Comment:
"by user" can be removed or changed to "by the user". Retirieve -> Retrieve . "query result after it completes". "while the query is still executing."